File: HappyPuppy/C#/RepositoryTests.cs
using System;
using NUnit.Framework;
using System.Collections.Generic;

[TestFixture]
public class RepositoryTests
{
    private ListBasedRepository repo;

    [SetUp]
    public void SetUp()
    {
         repo = new ListBasedRepository(CreateTestDatabase());   
    }

    [Test]
    public void retrieving_puppies()
    {
        Assert.That(repo.RetrievePuppies(), Is.EqualTo(
            new string[] { "Spike", "Pluto", "Daisy" }));
    }

    [Test]
    public void retrieving_owners()
    {
        Assert.That(repo.RetrieveOwners(), Is.EqualTo(
            new string[] { "Tom", "Jane" }));
    }

    [Test]
    public void removing_puppies()
    {
        repo.RemovePuppy("Daisy");

        Assert.That(repo.RetrievePuppies(), Has.No.Member("Daisy") );
    }

    [Test]
    public void adding_owners()
    {
        repo.AddOwner("Fernando");
        Assert.That(repo.RetrieveOwners(), Has.Member("Fernando") );
    }

    private Dictionary<string, List<string>> CreateTestDatabase()
    {
        Dictionary<string, List<string>> db = 
        new Dictionary<string, List<string>>();

        db.Add("owners", 
            new List<string>() { "Tom", "Jane" });

        db.Add("puppies", 
            new List<string>()
                { "Spike", "Pluto", "Daisy" });

        return db;
     }
    
}

File: HappyPuppy/Java/RepositoryTests.java
import java.util.*;
import org.junit.*;
import static org.hamcrest.CoreMatchers.*;
import static org.junit.Assert.*;

public class RepositoryTests
{
    private Repository.ListBasedRepository repo;

    @Before
    public void SetUp()
    {
         repo = new Repository.ListBasedRepository(CreateTestDatabase());   
    }

    @Test
    public void retrieving_puppies()
    {
        assertThat(repo.RetrievePuppies(), hasItems(
           "Spike", "Pluto", "Daisy"));
    }

    @Test
    public void retrieving_owners()
    {
        assertThat(repo.RetrieveOwners(), hasItems("Tom", "Jane"));
    }

    @Test
    public void removing_puppies()
    {
        repo.RemovePuppy("Daisy");

        assertThat(repo.RetrievePuppies(), not(hasItem("Daisy")));
    }

    @Test
    public void adding_owners()
    {
        repo.AddOwner("Fernando");
        assertThat(repo.RetrieveOwners(), hasItem("Fernando") );
    }

    private Map<String, List<String>> CreateTestDatabase()
    {
        Map<String, List<String>> db = 
            new HashMap<String, List<String>>();

        db.put("owners", new ArrayList<String>(
            Arrays.asList(new String[] { "Tom", "Jane" })));

        db.put("puppies", new ArrayList<String>(
            Arrays.asList(new String[] { "Spike", "Pluto", "Daisy" })));

        return db;
     }
}
